docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +76,24 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an array option.
+
+When `reservedFirst` is defined, React reserved props (`children`, `dangerouslySetInnerHTML`, `key`, `ref`) must be listed before all other props, but still respecting the alphabetical order:
```

Comment:
> React treats all lowercased names as a DOM element


I don't believe that's correct, you can name an imported composite component whatever you want and use that within JSX no problem; there's actually a rule in this repo to discourage that.

`jsx-pascal-case` (live-edit:) I realized that this rule must have the ability to determine user defined components! [I'll utilize the method found in that rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/jsx-pascal-case.js).

~~[Here's]( https://github.com/facebook/react/blob/master/src/test/ReactTestUtils.js?source=c#L155) how React Test Utils does it (it comes down to check [`ReactElement.isValidElement`](https://github.com/facebook/react/blob/b1b4a2fb252f26fe10d29ba60d85ff89a85ff3ec/src/isomorphic/classic/element/ReactElement.js#L376)). I'll work on it tomorrow and see what we have available on `node`.~~